src/Board.js

```diff
@@ -0,0 +1,157 @@
+                        <rect width="30" height="30" style={{ fill: "rgb(0,0,255)", strokeWidth: 0, stroke: "rgb(0,0,0)" }} />
+                        {displayBall(boardState[rowIndex][col])}
+                    </svg>
+                </Grid>
+            );
+        };
+
+        const getColumns = (keyPrefix, rowIndex) => {
+            const columns = [];
+            for (let col = 0; col < colCount; col++) {
+                columns.push(getColElement(keyPrefix, rowIndex, col));
+            };
+
+            return columns;
+        };
+
+        let getRowElement = (key, rowIndex) => (
+            <Grid className={'row-container'} xs={9} key={key}>
+                {getColumns(keyPrefix, rowIndex)}
+            </Grid>
+        );
+
+        for (let row = 0; row < rowCount; row++) {
+            result.push(getRowElement(`${keyPrefix}-${row}`, row));
+        };
+
+        return result;
+    };
+
+    const displayBall = (element) => {
+        if (element.isPlayerOneBall || element.isPlayerTwoBall) {
+
+            return (
+                element.isPlayerOneBall === true
+                    ? <SVGBallPlayer1 />
+                    : <SVGBallPlayer2 />
+            );
+        } else {
+
+            return (
+                <svg>
+                    <circle
+                        cx="12" cy="12" r="8"
+                        strokeWidth="6" stroke="white"
+                        fill="white"
+                    />
+                </svg >
+            );
+        };
+    };
+
+    const moveDownBall = (position, activeElement) => {
+        const copy = [...board];
+        let isTurnFinished = false;
+
+        let startLine = copy.length - 1;
+        let valueToSet = { isPlayerOneBall: false, isPlayerTwoBall: false };
+
+        while (startLine >= 0 && !isTurnFinished) {
+            if (getIsBallObjectEmpty(copy[startLine][position])) {
+                if (activeElement.isPlayerOneBall === true) {
+                    valueToSet = { isPlayerOneBall: true, isPlayerTwoBall: false };
+                    copy[startLine][position] = valueToSet;
+                } else {
+                    valueToSet = { isPlayerOneBall: false, isPlayerTwoBall: true };
+                    copy[startLine][position] = valueToSet;
+                }
+                isTurnFinished = true;
+            } else {
+                startLine--;
+            };
+        };
+
+        setBoard(copy);
+    };
+
+    const isLastCellFull = (position) => {
+        const { isPlayerOneBall, isPlayerTwoBall } = board[0][position];
+
+        return isPlayerOneBall || isPlayerTwoBall;
+    };
+
+    const renderInjectedComponent = () => {
+        let result = null;
+
+        if (props.getInjectedComponent) {
+            result = props.getInjectedComponent({
+                handleMoveDown: moveDownBall,
+                handleIsLastCellFull: isLastCellFull,
+                displayBoard: displayBoard
+            });
+        };
```

Comment:
Why not just pass the ball as a prop component directly, instead if using getInjectedComponent. This one is for Bobi